core/rsyslog/config.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package rsyslog
+
+import (
+	"crypto/tls"
+	"net/url"
+
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/cert"
+)
+
+var ErrNotConfigured = errors.Errorf("rsyslog fowarding not configured")
+
+func IsNotConfigured(err error) bool {
```

Comment:
I think we can drop this: Cause-handling is a client responsibility (Is funcs don't always do it...); and besides, Is funcs are only necessary when we're looking for error types rather than value. I'm finding that a pattern something like:

    err := Blah(blah, blah)
    switch errors.Cause(err) {
        case nil, ErrSafelyIgnorableInContext:
        case ErrNeedsReplacingSomehow:
            return ErrMoreAppropriateForOurClients
        default:
            return errors.Trace(err)
    }

...usually seems to fit pretty well.